### PR TITLE
Optimize subscribe performance

### PR DESCRIPTION
**Pull request by jgaskins**
`IO#gets(delimiter : String)` is slow, but `IO#gets(delimiter : Char)` is faster. I was able to achieve a ~43% improvement in subscriber throughput (1.86M vs 1.27M) by using a Char instead of a String delimeter.

This appears to be safe since there isn't anywhere that a `\n` could appear in either of these cases where a `\r` wouldn't precede it. So as long as the server is doing the right thing, we should be good.

### CODE REVIEW

**Comment by jgaskins** [comment]
Looks like I may have to base this branch on the 0.32-compatibility branch in order for specs to run

**Comment by jgaskins** [comment]
With 86b7b8c1cba951b584cc47934e216af5d6e3207a I'm getting almost 3x subscription throughput — 3.5M/sec up from 1.27M/sec.

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by derekcollison** [comment]
Will merge once we get the main one merged.

**Pull request merged**
